packages/core/src/browser/shell/application-shell.ts

```diff
@@ -659,6 +677,14 @@ export class ApplicationShell extends Widget {
         }
     }
 
+    protected registerViewPositions(area: ApplicationShell.Area, widgets: (Widget | undefined)[]): void {
+        for (const widget of widgets) {
+            if (widget) {
+                this.fireDidAddWidget(widget, area);
+            }
+        }
+    }
```

Comment:
I think this step is only necessary for the `main` and `bottom` areas. The `DockPanel.restoreLayout` method doesn't produce any `widgetAdded` events, but our implementation of `setLayoutData` for the side panels does already produce events.